scripts/install/setup.sh

```diff
@@ -1,21 +1,55 @@
 #!/usr/bin/env bash
 
-bold() {
-  echo ". $(tput bold)" "$*" "$(tput sgr0)";
-}
-
 err() {
   echo "$*" >&2;
 }
 
-~/spinnaker-for-gcp/scripts/manage/check_duplicate_dirs.sh || exit 1
-~/spinnaker-for-gcp/scripts/manage/check_git_config.sh || exit 1
+[ -n "$PARENT_DIR" ] || PARENT_DIR="$HOME"
+
+source $PARENT_DIR/spinnaker-for-gcp/scripts/manage/service_utils.sh
+
+check_for_required_binaries
+
+PARENT_DIR=$PARENT_DIR $PARENT_DIR/spinnaker-for-gcp/scripts/manage/check_git_config.sh || exit 1
 
-PROPERTIES_FILE="$HOME/spinnaker-for-gcp/scripts/install/properties"
+[ -z "$PROPERTIES_FILE" ] && PROPERTIES_FILE="$PARENT_DIR/spinnaker-for-gcp/scripts/install/properties"
 
 source "$PROPERTIES_FILE"
 
-~/spinnaker-for-gcp/scripts/manage/check_project_mismatch.sh
+check_for_shared_vpc $CI
+
+PARENT_DIR=$PARENT_DIR PROPERTIES_FILE=$PROPERTIES_FILE $PARENT_DIR/spinnaker-for-gcp/scripts/manage/check_project_mismatch.sh
+
+if [ "$CI" = true ]; then
+  OPERATOR_SA_EMAIL=$(gcloud config list account --format "value(core.account)")
+
+  SETUP_REQUIRED_ROLES=(cloudfunctions.developer compute.networkViewer container.admin iam.serviceAccountCreator iam.serviceAccountUser pubsub.editor redis.admin serviceusage.serviceUsageAdmin source.admin storage.admin)
```

Comment:
Having `roles/owner` would also be sufficient, yes?

Reply:
Not proposing we recommend this to the user, just pointing out that the script shouldn't fail if the service account has been granted just `roles/owner`.

Reply:
Yeah `roles/owner` would work. And with the change to remove the `"$CI " = true` check from this  code block, it is more likely and operator will run into this issue. I'll update.